processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -310,7 +331,10 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException();
                       }
-                      cursorOffset.increment();
+
+                      do {
+                        cursorOffset.increment();
+                      } while (!isDone() && !filterMatcher.matches());
```

Comment:
I'm worried about the performance implications of checking both isDone and filter on every call of advance. Can we do some benchmarking?